1. Importe numpy como 'np' e imprima o número da versão instalada no ambiente (local ou colab).

In [28]:
import numpy as np
np.version

<module 'numpy.version' from 'c:\\Users\\Mirella\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\numpy\\version.py'>

2. Crie um *array* 1D com números de 0 a 9 e imprima o resultado.

In [29]:
a = np.array(range(10))
print(a)
a

[0 1 2 3 4 5 6 7 8 9]


array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])

Crie uma matriz booleana de 3x3 e preencha os seus valores com 'True'. Imprima o resultado.
Dica: Verifique o uso da função full na documentação do numpy.

In [30]:
matriz_bool = np.full ((3, 3), True)
print(matriz_bool)
matriz_bool

[[ True  True  True]
 [ True  True  True]
 [ True  True  True]]


array([[ True,  True,  True],
       [ True,  True,  True],
       [ True,  True,  True]])

4. Crie um *array* 1D como números inteiros (50 elementos) de 0 a 1000 e **extraia** apenas os números pares. Imprima o resultado.

In [31]:
num_int = np.random.randint(0, 1000, 50)
print(num_int)
pares = num_int[num_int%2==0]
print(pares)

[ 67 955 639 352 634 157 428 807 792 715 907 270 904 836 353 722 281  16
 547 617 860 525 562 543 648 416 317 307 473 487 783 951 716 222 235 722
 207 443 362 187 460 910 409 738  78 557 418 370 142 371]
[352 634 428 792 270 904 836 722  16 860 562 648 416 716 222 722 362 460
 910 738  78 418 370 142]


5. Usando o *array* criando no exercício anterior, **substitua** todos os números ímpares maiores que 50 por -1 no *array* original. 

In [36]:
num50 = num_int[num_int > 50]
num50[num50 %2 != 0] = -1
num50

array([ -1,  -1,  -1, 352, 634,  -1, 428,  -1, 792,  -1,  -1, 270, 904,
       836,  -1, 722,  -1,  -1,  -1, 860,  -1, 562,  -1, 648, 416,  -1,
        -1,  -1,  -1,  -1,  -1, 716, 222,  -1, 722,  -1,  -1, 362,  -1,
       460, 910,  -1, 738,  78,  -1, 418, 370, 142,  -1])

6. Usando o *array* criando no exercício anterior, **substitua** todos os números ímpares com o valor -1 **sem** alterar o *array* original.

7. Crie um *array* 1D com 100 valores de ponto flutuante (**float**), converta-o em uma matriz 2D com 10 linhas e 10 colunas.

In [65]:
vetor_float = np.arange(100, dtype=float)
vetor_float.reshape((10,10))

array([[ 0.,  1.,  2.,  3.,  4.,  5.,  6.,  7.,  8.,  9.],
       [10., 11., 12., 13., 14., 15., 16., 17., 18., 19.],
       [20., 21., 22., 23., 24., 25., 26., 27., 28., 29.],
       [30., 31., 32., 33., 34., 35., 36., 37., 38., 39.],
       [40., 41., 42., 43., 44., 45., 46., 47., 48., 49.],
       [50., 51., 52., 53., 54., 55., 56., 57., 58., 59.],
       [60., 61., 62., 63., 64., 65., 66., 67., 68., 69.],
       [70., 71., 72., 73., 74., 75., 76., 77., 78., 79.],
       [80., 81., 82., 83., 84., 85., 86., 87., 88., 89.],
       [90., 91., 92., 93., 94., 95., 96., 97., 98., 99.]])

Crie duas matrizes 2D de inteiros aleatórios (de 0 a 100) com dimensões 10x10. Empilhe-as horizontalmente.
Dica: verique as funções hstack e concatenate na documentação do numpy

In [82]:
matriz1 = np.random.randint(0, 100, (10,10))
matriz2 = np.random.randint(0, 100, (10,10))
vertical = np.concatenate((matriz1, matriz2))
horizontal = np.hstack((matriz1, matriz2))
print(horizontal)


[[22 49 32 35 37 68 55 95 84 37 87 71 57 88 65 26 39 33 45 84]
 [46 99  7 40 35 29 25 17 21 51 34 68 48 21 83 66 21 16 37 36]
 [52 65  1 19 73 21 57  4 60 61 63 27 26 92 84 36 21 27 13 61]
 [42 16 52 18  1  2 65 71 47 60 81 68 42 58 91 77 67 78 44 21]
 [29 26 85 53 20 11 85 50  0 58 16 31 82  9 75 50 85 93 91 88]
 [74 41 98 99 79 66 50 98  4 16 81 48 93 10 73 60 53 71 15 42]
 [86 27 14 91 41 47 61 93 21 13 92 67 17  6 36 71 54 55 84 16]
 [87  5 92 30 55 44 77 57 16 36  8  8 93 96  2 79 11 84 41 38]
 [ 5 15 37 28 82 74 46 66 53 92 44 42 22 84  0 32 74 59 79 62]
 [95 42 67 24  6 54 20 28 95 30 73 47 68  4  1 78 92 33 41 48]]


Crie duas matrizes 2D de inteiros aleatórios (de 0 a 100) com dimensões 10x10. Empilhe-as verticalmente.
Dica: verique as funções vstack e concatenate na documentação do numpy

In [83]:
vertical

array([[22, 49, 32, 35, 37, 68, 55, 95, 84, 37],
       [46, 99,  7, 40, 35, 29, 25, 17, 21, 51],
       [52, 65,  1, 19, 73, 21, 57,  4, 60, 61],
       [42, 16, 52, 18,  1,  2, 65, 71, 47, 60],
       [29, 26, 85, 53, 20, 11, 85, 50,  0, 58],
       [74, 41, 98, 99, 79, 66, 50, 98,  4, 16],
       [86, 27, 14, 91, 41, 47, 61, 93, 21, 13],
       [87,  5, 92, 30, 55, 44, 77, 57, 16, 36],
       [ 5, 15, 37, 28, 82, 74, 46, 66, 53, 92],
       [95, 42, 67, 24,  6, 54, 20, 28, 95, 30],
       [87, 71, 57, 88, 65, 26, 39, 33, 45, 84],
       [34, 68, 48, 21, 83, 66, 21, 16, 37, 36],
       [63, 27, 26, 92, 84, 36, 21, 27, 13, 61],
       [81, 68, 42, 58, 91, 77, 67, 78, 44, 21],
       [16, 31, 82,  9, 75, 50, 85, 93, 91, 88],
       [81, 48, 93, 10, 73, 60, 53, 71, 15, 42],
       [92, 67, 17,  6, 36, 71, 54, 55, 84, 16],
       [ 8,  8, 93, 96,  2, 79, 11, 84, 41, 38],
       [44, 42, 22, 84,  0, 32, 74, 59, 79, 62],
       [73, 47, 68,  4,  1, 78, 92, 33, 41, 48]])

Para ler arquivos em formato txt que estão em formato de "tabelas", podemos usar a função genfromtxt() do numpy. Esta função recebe como argumentos: caminho do arquivo txt (path local ou uma url); delimiter para separar cada valor seguindo algum critério; dtype para converter os valores recebidos em algum tipo de dado conhecido (exemplo: float) e; usecols em que é definido o cabeçalho das colunas lidas a partir do arquivo.

Vejamos um exemplo de uso da função genfromtxt sobre a base de dados Iris.

In [33]:
import numpy as np
# Caminho da base de dados é uma url disponível na internet (visite o link abaixo
# pelo navegador para título de curiosidade :))
url = 'https://archive.ics.uci.edu/ml/machine-learning-databases/iris/iris.data'
# Lê os dados a partir do caminho passado e armazena em uma matriz 2D chamada iris_2d
iris_2d = np.genfromtxt(url, delimiter=',', dtype='float', usecols=[0,1,2,3])

In [34]:
# matriz 2D criada com os dados da base Iris
iris_2d

array([[5.1, 3.5, 1.4, 0.2],
       [4.9, 3. , 1.4, 0.2],
       [4.7, 3.2, 1.3, 0.2],
       [4.6, 3.1, 1.5, 0.2],
       [5. , 3.6, 1.4, 0.2],
       [5.4, 3.9, 1.7, 0.4],
       [4.6, 3.4, 1.4, 0.3],
       [5. , 3.4, 1.5, 0.2],
       [4.4, 2.9, 1.4, 0.2],
       [4.9, 3.1, 1.5, 0.1],
       [5.4, 3.7, 1.5, 0.2],
       [4.8, 3.4, 1.6, 0.2],
       [4.8, 3. , 1.4, 0.1],
       [4.3, 3. , 1.1, 0.1],
       [5.8, 4. , 1.2, 0.2],
       [5.7, 4.4, 1.5, 0.4],
       [5.4, 3.9, 1.3, 0.4],
       [5.1, 3.5, 1.4, 0.3],
       [5.7, 3.8, 1.7, 0.3],
       [5.1, 3.8, 1.5, 0.3],
       [5.4, 3.4, 1.7, 0.2],
       [5.1, 3.7, 1.5, 0.4],
       [4.6, 3.6, 1. , 0.2],
       [5.1, 3.3, 1.7, 0.5],
       [4.8, 3.4, 1.9, 0.2],
       [5. , 3. , 1.6, 0.2],
       [5. , 3.4, 1.6, 0.4],
       [5.2, 3.5, 1.5, 0.2],
       [5.2, 3.4, 1.4, 0.2],
       [4.7, 3.2, 1.6, 0.2],
       [4.8, 3.1, 1.6, 0.2],
       [5.4, 3.4, 1.5, 0.4],
       [5.2, 4.1, 1.5, 0.1],
       [5.5, 4.2, 1.4, 0.2],
       [4.9, 3

11. A partir dos dados lidos na matriz iris_2d, filtre as linhas que possuem (coluna 3) > 1.5 e (coluna 1) < 5.0.

12. A partir dos dados lidos na matriz iris_2d, calcule a **média** e **desvio padrão** de cada coluna presente nos dados. Imprima o resultado.

13. Crie um repositório no Github e versione todos os códigos que desenvolveram nesta lista de exercícios.